helper.sh

```diff
@@ -0,0 +1,128 @@
+#/bin/bash!
+
+# Color
+INFO='\033[38;5;29m'    # Dark Green
+TIP='\033[38;5;25m'     # Ocean Blue
+ERROR='\033[38;5;124m'  # Red
+END='\033[0m'       # No Color
+
+# Virables
+REPO_NAME=dv-web
+IMAGE_NAME=${REPO_NAME}
+# To avoid interfere with other projects in the future, when using helper running up the service, give it a isolated network
+NETWORK_NAME=dv_web_network
+
+# Text Block
+read -d '' instruction << EOM
+jq(lightweight command-line JSON processor) is not installed. If you want the script fully functional, please follow the instruction:
+Please install jq with following command:
+# brew install jq
+The waiting time to install it maybe a little bit longer than you expect
+EOM
+
+# Functions
+install_jq(){
+    # Check if the jq --version error output is zero length or not, if yes, then the jq is not installed
+    if [ -z "$(jq --version 2>/dev/null)" ];then
+        echo "${ERROR}$instruction${END}"
+    fi
+}
+
+# Build the image
+build_image(){
+    echo "\n${INFO}Build ${REPO_NAME} image...${END}"
+    docker build -f dev/Dockerfile -t ${REPO_NAME} --build-arg DEV_HOME=$(pwd) .
+}
+
+# Precheck image
+has_image(){
+    if [[ -z "$(docker images -q ${IMAGE_NAME})" ]];then
+        build_image
+    else
+        echo "${INFO}${IMAGE_NAME} already exists...${END}"
+    fi
+}
+
+# Create the network
+create_network(){
+    if [[ $(docker network inspect ${NETWORK_NAME} 2>/dev/null | jq '. | length') == 0 ]];then
+        echo "${INFO}Create network ${NETWORK_NAME}...${END}"
+        docker network create ${NETWORK_NAME} --driver=bridge
+    else
+        echo "${INFO}${NETWORK_NAME} already exists...${END}"
+    fi
+}
+
+start_app(){
+    if [[ -z "$(docker ps -aqf "name=${REPO_NAME}")" ]]; then
+        echo "${INFO}Start ${REPO_NAME}..." &&\
+        docker run --rm -it --name ${REPO_NAME} -v $(pwd):$(pwd) -p 3000:3000 --network ${NETWORK_NAME} ${IMAGE_NAME}
```

Comment:
There will be port confit if we try to start dv-ingestor and dv-web at the same time, usually, for different microservice, we will assign a different port.

Reply:
To start with, we can temporarily assign 3001 for frontend, because dv-ingestor is using 3000 node default port.
https://stackoverflow.com/questions/40714583/how-to-specify-a-port-to-run-a-create-react-app-based-project